src/creature.cpp

```diff
@@ -784,6 +803,8 @@ void Creature::drainHealth(Creature* attacker, int32_t damage)
 
 	if (attacker) {
 		attacker->onAttackedCreatureDrainHealth(this, damage);
+	} else {
+		lastHitCreatureId = 0;
```

Comment:
What's this for?

Reply:
To identify that something evil was responsible for your death, otherwise, if Pedro gives you a -1 hit point and then I do the command player:addHealth(-999999999999) in the description it will appear that Pedro was the one who kill, for the simple reason that he was the last character to hit you

Reply:
> To identify that something evil was responsible for your death

lol that's cute

Reply:
@ranisalt jajajajajajajaja 😂